### ISSUE TITLE
Loading data failed Expected all tensors to be on the same device,

### ISSUE THREAD

**Issue by umitmertcakmak** [weaviate/Verba#41]
Hey, I'm using Verba with OpenAI. 

I'm trying to import a txt file with Simple Reader and MiniLMEmbedding.

Receiving the following error:

```
✘ Loading data failed Expected all tensors to be on the same device,
but found at least two devices, cuda:0 and cpu! (when checking argument for
argument index in method wrapper_CUDA__index_select)
```

It seems to be a pytorch error related to moving model to device.

Here's what's intalled after pip install verba
`Successfully installed MarkupSafe-2.1.3 accelerate-0.24.1 aiohttp-3.8.6 aiosignal-1.3.1 annotated-types-0.6.0 anyio-4.0.0 async-timeout-4.0.3 asyncio-3.4.3 attrs-23.1.0 authlib-1.2.1 backoff-2.2.1 blis-0.7.11 catalogue-2.0.10 certifi-2023.7.22 cffi-1.16.0 charset-normalizer-3.3.2 click-8.1.7 cohere-4.33 confection-0.1.3 cryptography-41.0.5 cymem-2.0.8 fastapi-0.102.0 fastavro-1.8.2 filelock-3.13.1 frozenlist-1.4.0 fsspec-2023.10.0 goldenverba-0.3.0 h11-0.14.0 httptools-0.6.1 huggingface-hub-0.19.4 idna-3.4 importlib_metadata-6.8.0 jinja2-3.1.2 joblib-1.3.2 langcodes-3.3.0 mpmath-1.3.0 multidict-6.0.4 murmurhash-1.0.10 networkx-3.2.1 nltk-3.8.1 numpy-1.26.2 nvidia-cublas-cu12-12.1.3.1 nvidia-cuda-cupti-cu12-12.1.105 nvidia-cuda-nvrtc-cu12-12.1.105 nvidia-cuda-runtime-cu12-12.1.105 nvidia-cudnn-cu12-8.9.2.26 nvidia-cufft-cu12-11.0.2.54 nvidia-curand-cu12-10.3.2.106 nvidia-cusolver-cu12-11.4.5.107 nvidia-cusparse-cu12-12.1.0.106 nvidia-nccl-cu12-2.18.1 nvidia-nvjitlink-cu12-12.3.101 nvidia-nvtx-cu12-12.1.105 openai-0.27.9 packaging-23.2 pathy-0.10.3 pillow-10.1.0 preshed-3.0.9 psutil-5.9.6 pycparser-2.21 pydantic-2.5.1 pydantic-core-2.14.3 python-dotenv-1.0.0 pyyaml-6.0.1 regex-2023.10.3 requests-2.31.0 safetensors-0.4.0 scikit-learn-1.3.2 scipy-1.11.3 sentence-transformers-2.2.2 sentencepiece-0.1.99 smart-open-6.4.0 sniffio-1.3.0 spacy-3.6.1 spacy-legacy-3.0.12 spacy-loggers-1.0.5 srsly-2.4.8 starlette-0.27.0 sympy-1.12 thinc-8.1.12 threadpoolctl-3.2.0 tiktoken-0.5.1 tokenizers-0.15.0 torch-2.1.1 torchvision-0.16.1 tqdm-4.66.1 transformers-4.35.2 triton-2.1.0 typer-0.9.0 typing-extensions-4.8.0 urllib3-2.1.0 uvicorn-0.24.0.post1 uvloop-0.19.0 validators-0.21.0 wasabi-1.1.2 watchfiles-0.21.0 weaviate-client-3.23.1 websockets-12.0 yarl-1.9.2 zipp-3.17.0`

**Comment by thomashacker** [comment]
Thanks for the issue! I'll look into it, seems like we have a bug in the implementation

**Comment by thomashacker** [comment]
Fixed and merged on the `main` branch! It will be included in the `v0.3.1` release! You can install it by cloning the GitHub repo and using `pip install -e .` or wait for the release, which should come soon!
